docs/design/reserve-action.md

```diff
@@ -0,0 +1,254 @@
+scheduled requirement of the task now or later. This node can reserve resources for this task. If the idle resource of 
+the current node has met the scheduling condition of the task, resources will be allocated to this task, and the task 
+will be directly assigned to the node with `ForceAllocated` state. If the idle resource of the node can not meet the resource 
+requirement of the task, the node will reserve resources for this task, the task will be allocated to this node with 
+`FakeAllocated` state. Once a node reserve resources for a task, no other tasks will be scheduled to this node, for 
+the node's allocatable resources will no longer meet the scheduling requirements of any task.
+
+![reserve-logic](../../docs/images/reserve-logic.png)
+
+A node may reserve resources only for one task, or may reserve resources for multiple tasks, depending on the size of the 
+allocatable resources of node.
+
+#### The proportion of nodes participating in reserve action
+
+Because of that some of node resources was reserved for tasks and will not be used by other tasks, the resources 
+reservation will affect the task scheduling process of the entire cluster. In theory, the more nodes that participate in
+resource reservation, the better it is to solve the problem of the task being starved to death. However, the fewer nodes 
+participating in regular scheduling, and the greater the impact on regular cluster scheduling. Therefore, the number of 
+nodes participating in resource reservation is configurable.
+
+#### Node selection strategy to participate in reserve
+ 
+How to choose which nodes will be selected as candidate nodes for starving job scheduling. The process of selecting a 
+reserved node will include the following steps.
+
+1. Get all nodes in cluster
+2. PredicateNodes for task
+3. Calculating node score by idle resource and resource requirement of task
+4. Sort node by node score
+5. Re-define node score by threshold of the number of reserved nodes
+
+#### Node calculating score logic
+
+The process of calculating node scores includes two processes of `map` and `reduce`. In the `map` process, all nodes 
+calculated a score according to the idle resources of the node and resource requirement of task, and then all nodes are 
+sorted in descending order of rank.
+
+In the process of calculating node scores based on node idle resource, there will be `cpu` `memory` and `scalar resources` 
+participating in the score calculation. The formula for calculating score for a single resource is  
+
+<img src="http://chart.googleapis.com/chart?cht=tx&chl=\Large score=100-\frac{usedResource}{allocatableResource}" style="border:none;">
+
+The formula for calculating the node score based on the above three resources is
+
+<img src="http://chart.googleapis.com/chart?cht=tx&chl=\Large score=\frac{\frac{cpuScore%2b memoryScore}{2}%2b2*sclarScore}{3}" style="border:none;">
```

Comment:
What's the meaning of [LargScore?]

Reply:
`score` and `\Large` are separate here, and `\Large` is the formula syntax, indicating that the font of the following letters becomes larger

![image](https://user-images.githubusercontent.com/25945130/61192762-6266b600-a6e9-11e9-906f-2008816a2b67.png)

Reply:
It seems that the pic has not been fixed.

Reply:
Is there any reference for this formula?

Reply:
What about this question?

Reply:
The formula here uses the syntax of html, calls the api of google chart, generates the mathematical formula of Latex, and can be seen by tool of markdown.
referenced this document
https://developers.google.com/chart/image/docs/making_charts
![image](https://user-images.githubusercontent.com/25945130/61341353-77be1a80-a878-11e9-9c97-dd21c9a4eace.png)

Reply:
Is there any reference for this formula?

Reply:
updated to add reference for GoogleChart quota in `References`

Reply:
let's save the pic locally :)

Reply:
I will replace it

Reply:
done